### PR TITLE
fix(unroll): correct per-tile HBM byte stride for tiled dimensions

### PR DESCRIPTION
**Pull request by marnold-ibm**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [x] bug
- [ ] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

 Fixes three related bugs that caused wrong HBM base-address advances when coarse-tiling over a non-outermost host dimension (e.g. H in [B,H,Lq,D] with B=1).

**ROOT CAUSE**

Three invariants were broken:

1. scheduler.py (_tiled_syms_for_sched_node_at_depth): loop_tiled_dims stores host-range indices (e.g. 1 for H in [B=1,H,Lq,D]), but was used directly as iteration-space  key indices. Since B=1 is dropped from the iteration space, H is at index 0, not 1 — the wrong symbol (Lq) was selected as the tiled symbol.

2. spyre_kernel.py (simplify_op_spec): align_tensors rewrites device_coordinates to rearrange how iteration variables map to device dimensions, but didn't update stride_map to match. After the rewrite, stride_map[d] no longer corresponded to coordinate d, so the unroller read the wrong host stride.

3. unroll.py (_byte_stride_for_arg): relied on stride_map[d] being correct (which broke due to (2) above).

**FIX**

1. In _tiled_syms_for_sched_node_at_depth: walk data.ranges to build a host-index → iteration-space-index mapping, skipping unit-size dims.

2. In simplify_op_spec: after align_tensors reshuffles device_coordinates, remap stride_map entries to stay consistent — for each new coordinate position, find the old  position with the same symbol and carry its stride forward.

3. _byte_stride_for_arg now works correctly because stride_map is kept consistent.

Reproducer (fails with error ~9 before fix, passes after):
 `q * v with spyre_hint(slices={"H": 2}) on [B=1, H=8, Lq=256, D=64]`

#### Which issue(s) this PR is related to:

Part of #1358 

#### Additional note:
Standalone version of pytest  `test_hint_h_tiling_elementwise`
```
import math
import torch
from torch_spyre._inductor import config, spyre_hint
import torch_spyre._inductor.propagate_named_dims as pnd

declare_tensor_dim = pnd.declare_tensor_dim
name_tensor_dims = pnd.name_tensor_dims

torch.manual_seed(42)

# Small enough to inspect
B, H, Lq, Lk, D = 1, 8, 256, 256, 64
# NOTE: H must be divisible by H_TILES

Q = torch.randn(B, H, Lq, D, dtype=torch.float16)
K = torch.randn(B, H, Lk, D, dtype=torch.float16)
V = torch.randn(B, H, Lk, D, dtype=torch.float16)

H_TILES = 2 # change to 1 to compare

def attn(q, v):
    with spyre_hint(slices={"H": H_TILES}):
        # pointwise + single matmul — fewest ops that still compile
        out = q * v
    return out

# CPU reference
ref = attn(Q, V)

# Device
Q_d = Q.to("spyre")
V_d = V.to("spyre")

declare_tensor_dim("B", B)
declare_tensor_dim("H", H)
declare_tensor_dim("Lq", Lq)
declare_tensor_dim("Lk", Lk)
declare_tensor_dim("D", D)
name_tensor_dims(Q_d, ["B", "H", "Lq", "D"])
name_tensor_dims(V_d, ["B", "H", "Lk", "D"])

config.coarse_tiling = True

result = torch.compile(attn)(Q_d, V_d).cpu()

print(f"H_TILES={H_TILES}")
print(f"ref:    {ref.flatten()[:8]}")
print(f"result: {result.flatten()[:8]}")
diff = torch.abs(ref - result)
print(f"max abs diff: {diff.amax().item()}")
print(f"per-head max: {diff.amax(dim=(0,2,3))}")

torch.testing.assert_close(
    result, ref, equal_nan=True, atol=0.02, rtol=0.1,
    msg=lambda msg: f"compiled spyre <-> cpu mismatch\n\n{msg}\n",
)
print("PASSED")
```

### CODE REVIEW

**Review by dgrove-oss — APPROVED** [review]
lgtm
